torch/distributed/distributed_c10d.py

```diff
@@ -3308,6 +3312,16 @@ def reduce_scatter_tensor(output, input, op=ReduceOp.SUM, group=None, async_op=F
     opts.reduceOp = op
 
     group = group or _get_default_group()
+    if get_backend(group) == "gloo":
```

Comment:
@H-Huang @kwen2501 I'd like to be able to fallback for these collectives when I have gloo backend, to make FSDP work on CPUs. Any thoughts on how to achieve this and if @kumpera suggestion sounds good?